tests/integration/tests/test_clustering.py

```diff
@@ -88,9 +88,42 @@ def test_no_remove(instances: List[harness.Instance]):
     assert "control-plane" in util.get_local_node_status(joining_cp)
     assert "worker" in util.get_local_node_status(joining_worker)
 
-    nodes = util.ready_nodes(cluster_node)
-
     cluster_node.exec(["k8s", "remove-node", joining_cp.id])
+    nodes = util.ready_nodes(cluster_node)
     assert len(nodes) == 3, "cp node should not have been removed from cluster"
     cluster_node.exec(["k8s", "remove-node", joining_worker.id])
+    nodes = util.ready_nodes(cluster_node)
     assert len(nodes) == 3, "worker node should not have been removed from cluster"
+
+
+@pytest.mark.node_count(2)
+def test_join_with_non_hostname_name(instances: List[harness.Instance]):
+    cluster_node = instances[0]
+    joining_cp = instances[1]
+
+    out = cluster_node.exec(
+        ["k8s", "get-join-token", "my-token"],
+        capture_output=True,
+        text=True,
+    )
+    join_token = out.stdout.strip()
+
+    join_config = """
+extra-sans:
+- my-token
+"""
+    joining_cp.exec(
+        ["k8s", "join-cluster", join_token, "--name", "my-node", "--file", "-"],
```

Comment:
We should also test not supplying a `--name` flag at all which is what we do in capi. The command will use the `hostname` if not supplied.